build_runner/lib/src/asset_graph/serialization.dart

```diff
@@ -30,12 +30,12 @@ class _AssetGraphDeserializer {
         _deserializeDigest(_serializedGraph['buildActionsDigest'] as String),
         _serializedGraph['dart_version'] as String);
 
-    var packageNames = _serializedGraph['packages'] as List<String>;
+    var packageNames = _serializedGraph['packages'] as List;
```

Comment:
should we `.cast<String>()` here and remove the `as String` below?

Reply:
Same-same, right? The cast call feels "heavier" – @natebosch ?

Reply:
I am curious what effective dart says here now - its effectively the same though I think since its a lazy cast

Reply:
@munificent ?

Reply:
Ah, "Cast the elements on access. If you immediately iterate over the collection, cast each element inside the iteration.", so LGTM